rotkehlchen/chain/evm/accounting/interfaces.py

```diff
@@ -97,13 +98,28 @@ def _process_deposit_or_withdrawal(
         # Consume the events
         events_consumed = 1  # counting the event that started it
         for idx in range(events_to_consume):
-            next_event = next(other_events, None)
-            if next_event is None:
+            if (next_event := other_events.peek(None)) is None:
                 log.debug(f'Could not consume event nr. {idx} for {event.counterparty} deposit/withdrawal')  # noqa: E501
                 return events_consumed
+            elif next_event.event_identifier != event.event_identifier:
+                # ensure that we only process entries of the same event group.
+                log.error(
+                    'In _process_deposit_or_withdrawal the next event belongs to a diffetent '
+                    f'group when processing {event}. Next event is {next_event}. Skipping',
+                )
+                return events_consumed
 
-            events_consumed += 1
+            if next_event.entry_type != event.entry_type:
+                # ensure that we can't mix evm events with other types of events.
+                log.error(
+                    f'Expected to consume an event of type {event.entry_type} but the next event'
+                    f'is of type {next_event.entry_type}. {event=}, {next_event=}. Skipping...',
```

Comment:
Be consistent between your log entry wording

```suggestion
                    f'is of type {next_event.entry_type}. {event=}, {next_event=}. Skipping',
```